### PR TITLE
test: refactor test-assortativity for readability

### PR DESCRIPTION
**Pull request by maelle**
@szhorvat does the name `reference_assortativity()` make sense for that test helper function?

### CODE REVIEW

File: tests/testthat/test-assortativity.R [hunk 1]
```diff
@@ -13,29 +13,27 @@ test_that("assortativity works", {
...
-  as2 <- assR(g)
+  assortativity_igraph <- assortativity(
+    g,
+    values = degree(g, mode = "out"),
+    values.in = degree(g, mode = "in")
+  )
 
-  expect_that(asd, equals(as))
-  expect_that(asd, equals(as2))
+  expect_equal(assortativity_degree(g), assortativity_igraph)
```
Comment (maelle) [inline tests/testthat/test-assortativity.R:22]: is it bad to run `assortativity_degree(g)` twice? I find that more readable than giving it a name, that the reader has to keep track of.

**Comment by szhorvat** [comment]
That name is no worse or better than the previous to me, use whatever you like. The function computes degree assortativity in a different way—it's not a baseline or a reference, it's just a "different" way (but really the same as what the C core does). Perhaps this sort of renaming is not the best use of our time. This is a trivial test that just repeats what's already being done in the C core. Users would be much better served by exposing new functionality, fixing bugs, and cleaning up the interface.

**Comment by maelle** [comment]
I am going through all test files as part of the test refactoring efforts. :wink:

File: tests/testthat/test-assortativity.R (hunk 1 above)
Comment (szhorvat) [reply tests/testthat/test-assortativity.R:22]: Running twice is fine.

**Review by szhorvat — APPROVED** [review]

**Pull request merged**